src/Service.Host/client/src/components/parts/Part.js

```diff
@@ -34,15 +34,214 @@ function Part({
     privileges,
     userName,
     userNumber,
-    options,
+    templateName,
     partTemplates,
     liveTest,
     fetchLiveTest,
     fetchParts,
     partsSearchResults
 }) {
+    const defaultPart = {
+        partNumber: '',
+        description: '',
+        accountingCompany: 'LINN',
+        psuPart: false,
+        stockControlled: true,
+        cccCriticalPart: false,
+        safetyCriticalPart: false,
+        paretoCode: 'U',
+        createdBy: userNumber,
+        createdByName: userName,
+        dateCreated: new Date(),
+        railMethod: 'POLICY',
+        preferredSupplier: 4415,
+        preferredSupplierName: 'Linn Products Ltd',
+        qcInformation: '',
+        qcOnReceipt: false,
+        orderHold: false
+    };
     const creating = () => editStatus === 'create';
+
+    // all updates to state are now enacted by dispatching actions
+    // and processing them with this reducer, just like redux does
+    function partReducer(state, action) {
+        switch (action.type) {
+            case 'initialise':
+                if (creating()) {
+                    return { ...state, part: defaultPart };
+                }
+                return { ...state, part: action.payload, prevPart: action.payload };
+            case 'loadTemplate':
+                return { ...state, part: action.payload };
+            case 'fieldChange':
+                if (action.fieldName === 'rawOrFinished') {
```

Comment:
Yeah,

> I wonder about testing as well. I guess if its in the component it should be tested by the interactions rather than testing the reducer itself?

I guess so. I thought about trying to extract the reducer to a separate file. But it relies on having access to props which makes that tricky. I actually wonder whether relying on external things is allowed in the world of pure reducer functions now that I think about it more. I guess I could pass any prop date in attached to the actions that are dispatched